esmvaltool/diag_scripts/kcs/global_matching.py

```diff
@@ -0,0 +1,187 @@
+    return [year - 14, year + 15], target_dt
+
+
+def _timeline(axes, yloc, interval):
+    """Plot an interval near the bottom of the plot."""
+    xmin, xmax = interval
+
+    # Later years should be located slightly higher:
+    # yloc is relative to the axes, not in data coordinates.
+    yloc = 0.05 + yloc / 20
+
+    plot_args = dict(transform=axes.get_xaxis_transform(),
+                     linewidth=2,
+                     color='red')
+
+    axes.plot([xmin, xmax], [yloc] * 2, **plot_args, label='Selected periods')
+    axes.plot([xmin] * 2, [yloc - 0.01, yloc + 0.01], **plot_args)
+    axes.plot([xmax] * 2, [yloc - 0.01, yloc + 0.01], **plot_args)
+
+
+def make_plot(metadata, scenarios, cfg):
+    """Make figure 3, left graph.
+
+    Multimodel values as line, reference value in black square,
+    steering variables in dark dots.
+    """
+    fig, axes = plt.subplots()
+    for member in select_metadata(metadata, variable_group='tas_cmip'):
+        filename = member['filename']
+        dataset = xr.open_dataset(filename)
+        if 'MultiModel' not in filename:
+            axes.plot(dataset.time.dt.year,
+                      dataset.tas.values,
+                      c='grey',
+                      alpha=0.3,
+                      lw=.5,
+                      label='CMIP members')
+        else:
+            # Only display stats for the future period:
+            dataset = dataset.sel(time=slice('2010', None, None))
+            axes.plot(dataset.time.dt.year,
+                      dataset.tas.values,
+                      color='k',
+                      linewidth=2,
+                      label='CMIP ' + Path(filename).stem.split('_')[0][10:])
+
+    for member in select_metadata(metadata, variable_group='tas_target'):
+        filename = member['filename']
+        dataset = xr.open_dataset(filename)
+        if 'MultiModel' not in filename:
+            axes.plot(dataset.time.dt.year,
+                      dataset.tas.values,
+                      color='blue',
+                      linewidth=1,
+                      label=member['dataset'])
+
+    # Add the scenario's with dots at the cmip dt and bars for the periods
+    for i, scenario in enumerate(scenarios):
+        axes.scatter(scenario['year'],
+                     scenario['cmip_dt'],
+                     s=50,
+                     zorder=10,
+                     color='r',
+                     label=r"Scenarios' steering $\Delta T_{CMIP}$")
+        _timeline(axes, i, scenario['period_bounds'])
+
+    handles, labels = plt.gca().get_legend_handles_labels()
+    by_label = dict(zip(labels, handles))  # dict removes dupes
+    axes.legend(by_label.values(), by_label.keys())
+    axes.set_xlabel('Year')
+    axes.set_ylabel(r'Global mean $\Delta T$ (K) w.r.t. reference period')
+
+    # Save figure
+    filename = get_plot_filename('global_matching', cfg)
+    fig.savefig(filename, bbox_inches='tight', dpi=300)
```

Comment:
It would be good to also add provenance for the plot

Reply:
Done